cookbooks/aws-parallelcluster-config/recipes/fsx_mount.rb

```diff
@@ -16,15 +16,21 @@
 # limitations under the License.
 
 fsx_fs_id_array = node['cluster']['fsx_fs_ids'].split(',')
+fsx_fs_type_array = node['cluster']['fsx_fs_types'].split(',')
 fsx_shared_dir_array = node['cluster']['fsx_shared_dirs'].split(',')
 fsx_dns_name_array = node['cluster']['fsx_dns_names'].split(',')
 fsx_mount_name_array = node['cluster']['fsx_mount_names'].split(',')
+fsx_volume_junction_path_array = node['cluster']['fsx_volume_junction_paths'].split(',')
 # Check to see if FSx is created
 fsx_fs_id_array.each_with_index do |fsx_fs_id, index|
   fsx_shared_dir = fsx_shared_dir_array[index]
   fsx_dns_name = fsx_dns_name_array[index]
+  fsx_fs_type = fsx_fs_type_array[index]
+  fsx_volume_junction_path = fsx_volume_junction_path_array[index]
+
   # Path needs to be fully qualified, for example "shared/temp" becomes "/shared/temp"
   fsx_shared_dir = "/#{fsx_shared_dir}" unless fsx_shared_dir.start_with?('/')
+  fsx_volume_junction_path = "/#{fsx_volume_junction_path}" unless fsx_volume_junction_path.nil? || fsx_volume_junction_path.start_with?('/')
```

Comment:
Here you are assuming that `fsx_volume_junction_path` is *nil* for FsxLustre. 
Isn't it *empty* instead of *nil*?

Reply:
If `fsx_volume_junction_paths` is empty string, the result of the `split` is nil. Therefore, the check here is necessary to avoid type mismatch exception.
If it is empty string, we will prepend `/` too.